Userland/Applications/TerminalSettings/TerminalSettingsWidget.cpp

```diff
@@ -181,6 +181,52 @@ TerminalSettingsViewWidget::TerminalSettingsViewWidget()
     // whether that was filled in by the above defaulting code or by the user.
     use_default_font_button.set_checked(m_font == Gfx::FontDatabase::the().default_fixed_width_font(), GUI::AllowCallback::No);
     font_selection.set_enabled(!use_default_font_button.is_checked());
+
+    auto& terminal_cursor_block = *find_descendant_of_type_named<GUI::RadioButton>("terminal_cursor_block");
+    auto& terminal_cursor_underline = *find_descendant_of_type_named<GUI::RadioButton>("terminal_cursor_underline");
+    auto& terminal_cursor_bar = *find_descendant_of_type_named<GUI::RadioButton>("terminal_cursor_bar");
+
+    auto& terminal_cursor_blinking = *find_descendant_of_type_named<GUI::CheckBox>("terminal_cursor_blinking");
+
+    m_cursor_shape = VT::TerminalWidget::parse_cursor_shape(Config::read_string("Terminal", "Cursor", "Shape"));
+    m_original_cursor_shape = m_cursor_shape;
+
+    switch (m_cursor_shape) {
+    case VT::CursorShape::Block:
+        terminal_cursor_block.set_checked(true);
+        break;
+    case VT::CursorShape::Underline:
+        terminal_cursor_underline.set_checked(true);
+        break;
+    case VT::CursorShape::Bar:
+        terminal_cursor_bar.set_checked(true);
+        break;
+    default:
+        terminal_cursor_block.set_checked(true);
```

Comment:
I made `parse_cursor_shape()` return `CursorShape::None` if the config string for shape is neither "Block", "Bar" or "Underline", so `default` is hit if an unspecified string is passed (consequently we could change that `default` to `CursorShape::None` but I think the current version indicates better what it does). I thought disabling cursor in terminal as a setting is pointless so it uses the `None` for indication of when we should use a default value.

Reply:
> so default is hit if an unspecified string is passed

Could you make `parse_cursor_shape()` return an `Optional<CusorShape>` instead? That's generally what we do for most other parsing APIs.

Reply:
Even with changed signature of this function, we will still have to account for the `None` shape in the `switch` statement which `default` handles well, otherwise the compiler will complain.

Reply:
Changed this `parse_cursor_shape()` function. This fragment now looks something like this.

```cpp
    m_cursor_shape = VT::TerminalWidget::parse_cursor_shape(Config::read_string("Terminal", "Cursor", "Shape")).value_or(VT::CursorShape::Block);
    m_original_cursor_shape = m_cursor_shape;

    switch (m_cursor_shape) {
    case VT::CursorShape::Underline:
        terminal_cursor_underline.set_checked(true);
        break;
    case VT::CursorShape::Bar:
        terminal_cursor_bar.set_checked(true);
        break;
    default:
        terminal_cursor_block.set_checked(true);
```